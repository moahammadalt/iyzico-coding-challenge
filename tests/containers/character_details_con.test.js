import React, { Component } from 'react';
import { BrowserRouter as Router, Link } from "react-router-dom";
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CharacterDetailsContainer from '../../src/containers/character_details_con';
import BackButton from '../../src/components/back_button';

const mockStore = configureStore();
  
const initialState = {
  loadingSpinner: false,
  selectedCharacter: {}
};

const setInitialState = state => state ? {...initialState, ...state} : {...initialState};

describe('First scenario: when the character details page first opens', () => {

  it('should render loading spinner div when loadingSpinner prop is changed', () => {
    
    const store = mockStore(setInitialState({loadingSpinner: true}));

    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <CharacterDetailsContainer />
        </Provider>
      </Router>
    );

    expect(wrapper.find('.loading').exists()).toBeTruthy();
  });

  it('should render back button componenet', () => {
    
    const store = mockStore(setInitialState());

    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <CharacterDetailsContainer />
        </Provider>
      </Router>
    );

    expect(wrapper.find(BackButton).exists()).toBeTruthy();
  });

  it('should render CharacterDetails componenet when selectedCharacter prop is not empty', () => {
    
    const store = mockStore(
      setInitialState({
        selectedCharacter: {
          name: 'test'
        }
      })
    );

    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <CharacterDetailsContainer />
        </Provider>
      </Router>
    );

    expect(wrapper.find(BackButton).exists()).toBeTruthy();
  });

});



// Snapshots
describe('<CharacterDetailsContainer /> Snapshot test for character details components', () => {

  it('Should match the CharacterDetailsContainer snapshot before character data loading', () => {

    const store = mockStore(setInitialState());

    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <CharacterDetailsContainer />
        </Provider>
      </Router>
    );

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should match the CharacterDetailsContainer snapshot after character data loading', () => {

    const store = mockStore(
      setInitialState({
        selectedCharacter: {
          name: 'test'
        }
      })
    );

    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <CharacterDetailsContainer />
        </Provider>
      </Router>
    );

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });

});