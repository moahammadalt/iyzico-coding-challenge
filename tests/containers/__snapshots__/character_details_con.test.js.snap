// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterDetailsContainer /> Snapshot test for character details components Should match the CharacterDetailsContainer snapshot after character data loading 1`] = `
<div
  className="character-container"
>
  <div
    className="back-btn-container col-md-1"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="sorry, not available now"
        src="test-file-stub"
      />
    </a>
  </div>
  <div
    className="col-md-11"
  >
    <div
      className="col-md-6"
    >
      <img
        alt="sorry, not available now"
        className="character-img-left rounded"
      />
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="character-details-right"
      >
        <div
          className="character-detail-field border-bottom row"
        >
          <label
            className="col-sm-4"
          >
            NAME:
          </label>
          <p
            className="col-sm-8"
          >
            test
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CharacterDetailsContainer /> Snapshot test for character details components Should match the CharacterDetailsContainer snapshot before character data loading 1`] = `
<div
  className="character-container"
>
  <div
    className="back-btn-container col-md-1"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="sorry, not available now"
        src="test-file-stub"
      />
    </a>
  </div>
</div>
`;
